codeinterpreterapi/session.py

```diff
@@ -80,13 +104,25 @@ def _agent(self) -> BaseSingleActionAgent:
         )
 
     def _agent_executor(self) -> AgentExecutor:
+        if self.history_storage == "file":
+            history = FileChatMessageHistory(
+                f"{os.environ['SESSION_FOLDER_STORAGE']}/{self.session_id}.json"
+            )
+        elif self.history_storage == "redis":
+            history = RedisChatMessageHistory(
```

Comment:
I really like this idea! Maybe this can be done with the option to input your own history backend when creating the session.

Reply:
yes, it is my target, if the history exists it will be loaded, otherwise, the new empty one will be created. It is convenient to check the problem when an error is occupied.